packages/ng-query/src/lib/infinite-query.ts

```diff
@@ -1,11 +1,111 @@
 import { inject, Injectable, InjectionToken } from '@angular/core';
+import {
+  QueryKey,
+  QueryFunctionContext,
+  InfiniteQueryObserverOptions,
+  QueryObserver,
+  notifyManager,
+  InfiniteQueryObserver,
+} from '@tanstack/query-core';
+import { Observable, Subscription, take, tap } from 'rxjs';
 
 import { QueryClient } from './query-client';
+import { NgInfiniteQueryObserverResult } from './types';
 
 @Injectable({ providedIn: 'root' })
 class InfiniteQuery {
   private instance = inject(QueryClient);
-  use() {}
+  use<TQueryFnData, TError = unknown, TData = TQueryFnData>(
+    queryKey: QueryKey,
+    queryFn: (context: QueryFunctionContext<QueryKey>) => Observable<TData>,
+    options?: Omit<
+      InfiniteQueryObserverOptions<TQueryFnData, TError, TData, QueryKey>,
+      'queryKey' | 'queryFn'
+    >
+  ): Observable<NgInfiniteQueryObserverResult<TData, TError>> & {
+    instance: QueryObserver<TQueryFnData, TError, TData, QueryKey>;
+  } {
+    const defaultedOptions = this.instance.defaultQueryOptions(options);
+
+    defaultedOptions._optimisticResults = 'optimistic';
```

Comment:
I see that it's the same code as the basic query expect the observer. Can you please extract it to a reusable function?

Reply:
done ✅ let me know if the abstraction makes sense